### DISCUSSION TITLE
Question about fine-tune the pre-trained model.

### DISCUSSION THREAD

**Discussion by turbosonics** [MDIL-SNU/SevenNet#115] [Q&A] [answered]
Hi devs, I have a general question about fine-tune of SevenNet's Materials Project pretrained model. 

1. Must I need to use fine_tune.yaml to fine-tune the SevenNet MP pretrained model? 
I know I can't change some parameters in model section of yaml files. But can I change some parameters like self_connection_type, parameters under train/continue section, and/or batch_size for such fine-tune training of Materials Project pretrained model? 

2. Between xyz format and sevennet graph format, which one is better for such fine tuning? 
My AIMD training data consist of 20k frames. I wonder if converting this to sevennet graph would bring faster training or not. (With cutoff 5.0 to match with pre-trained model). I guess the new version no longer print out *.sevenn_data file, instead I can see ase.pt file in new folder. Is this correct? 

3. Is there any benchmark test plot for training time vs GPU for given training condition? 
Scaled-size test from your group's paper is about LAMMPS MD simulation, not about training. But I wonder if there is data about GPU number vs training speed/efficiency somewhere. 

Thanks.

**Comment by hexagonrose** [comment, marked as answer]
Hello @turbosonics 
1. fine_tune.yaml file is an example preset of fine-tuning, so you can change the parameters that you want. At sevenn/presets/base.yaml file, there are some annotations for each tags.

2. For training, file formats such as xyz or OUTCAR should be converted into sevennet graph data format (.pt). If ase.pt file is generated in sevenn_data folder after data preprocessing, I think it is the sevennet graph format of the dataset you want to train. So, if you delete .pt file, sevennet trainer will do data preprocessing to make .pt file again.

3. We don't have the graph about GPU number vs training speed/efficiency, but it'll ideally be faster as the number of GPUs increases.

If you have any other questions, feel free to ask us. Thank you!

**Comment by YutackPark** [comment]
For more about 2.

It is correct. Old `.sevenn_data` file is more bloated and therefore inefficient. SevenNet still can read both of them but I recommend to use `.pt` file, as it is written using `TorchGeometric`, therefore more memory efficient and slightly faster. Between xyz files and these processed files, there is no difference in training speed. Using `sevenn_graph_build` is just a separation of preprocessing step (graph build) from training. If `xyz` file is directly given to `sevenn`, it does the same thing (build graph) internally, and there is no difference in training speed.
